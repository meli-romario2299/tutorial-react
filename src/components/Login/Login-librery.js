import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import { useForm } from "react-hook-form";
import "./Login.css";

const Login = ({ backgroundColor }) => {
  console.log('miComponente renderizado');


  const [form, setForm] = useState({
    email: 'miemail@gmail.com',
    password: '12345'
  })

  let navigate = useNavigate();
  const handleLogin = () => {
    navigate('/dashboard');
  }

  const { register, handleSubmit, watch } = useForm();

  const miFuncion = (data) => console.log(data);

  return (
    <>
      <div id="login" style={{backgroundColor}}>
        Login
      </div>
      <form onSubmit={handleSubmit(miFuncion)} >
        <input
          placeholder="Ingresa tu email"
          type="email"
          name="email"
          {...register("email")}
        />
        <br />
        <input
          placeholder="Ingresa tu password"
          type="text"
          name="password"
          {...register("password")}
          />
        <br />
        <button type="submit">Login</button>
      </form>
    </>
  );
};

export default Login;
