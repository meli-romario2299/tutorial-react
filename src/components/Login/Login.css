#login {
  width: 100%;
  height: 70px;
  background-color: red;
  text-align: center;
}

form {
  text-align: center;
  margin-top: 10px;
}