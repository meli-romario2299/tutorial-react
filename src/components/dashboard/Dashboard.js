import React, { useEffect, useState } from 'react'
import { useNavigate } from "react-router-dom";

const Dashboard = () => {
  
  // console.log('Componente Dashboard Renderizado');
  
  const [resultado, setResultado] = useState(10);
  
  let navigate = useNavigate();
  
  const handleSumar = () => {
    setResultado(resultado+1);
  };

  useEffect(() => {
    console.log('Efecto se ejecuta por primera y única vez');
    // return () => {
    //   console.log('Hola, estoy desapareciendo');
    // }
  }, []);

  useEffect(() => {
    console.log('Efecto del resultado se está ejecutando...');
  }, [resultado]);
  
  const handleBack = () => {
    navigate('/');
  }

  return (
    <>
    <div>Calculadora</div>
    <hr />
    <button onClick={handleSumar}>Sumar +1</button>
    <div>Resultado: {resultado}</div>
    <button onClick={handleBack}>Back</button>
    </>
  )
}

export default Dashboard