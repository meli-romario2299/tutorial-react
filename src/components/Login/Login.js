import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import { string } from "prop-types"

import { useForm } from "../../hooks/useForm";

import "./Login.css";

const Login = ({ backgroundColor }) => {
  
  console.log('Componente Login Renderizado');
  
  const [form, setForm] = useState({
    email: 'miemail@gmail.com',
    password: '12345'
  })
  
  const { handleSubmit, handleChangeInput } = useForm({ form, setForm });

  let navigate = useNavigate();
  
  const handleLogin = () => {
    navigate('/dashboard');
  }

  return (
    <>
      <div id="login" style={{backgroundColor}}>
        View de Login
      </div>
      <form onSubmit={(e) => handleSubmit(e, handleLogin)} >
        <input
          placeholder="Ingrese su email corporativo"
          type="email"
          name="email"
          value={form.email}
          onChange={handleChangeInput}
        />
        <br />
        <input
          placeholder="Ingresa tu password"
          type="text"
          name="password"
          value={form.password}
          onChange={handleChangeInput}
          />
        <br />
        <button type="submit">Login</button>
      </form>
    </>
  );
};

Login.propTypes = {
  backgroundColor: string.isRequired
};

export default Login;
