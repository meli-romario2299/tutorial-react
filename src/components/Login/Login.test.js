import { fireEvent, render, screen } from '@testing-library/react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import '@testing-library/jest-dom';

import Login from './Login';
import Dashboard from '../dashboard/Dashboard';

describe("Prueba del componente de Login", () => {

  beforeEach(() => {
    
  })


  test('Verificar si el componente se encuentra renderizado', () => { 

    const propierties = {
      backgroundColor: 'red'
    }

    const { container } = render(
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Login backgroundColor={propierties.backgroundColor} />} />
          {/* <Route path="dashboard" element={<Dashboard />} /> */}
      </Routes>
      </BrowserRouter>
    );
    expect( screen.getByText('View de Login') ).toBeInTheDocument();
    expect( container ).toMatchSnapshot();

   })

   test("Testeando la funcionalidad del formulario", () => {

    const propierties = {
      backgroundColor: 'red'
    }

    const { container } = render(
      <BrowserRouter>
        <Routes>
          <Route path="/" element={<Login backgroundColor={propierties.backgroundColor} />} />
          <Route path="dashboard" element={<Dashboard />} />
      </Routes>
      </BrowserRouter>
    );

    fireEvent.change(
      screen.getByPlaceholderText('Ingrese su email corporativo'),
      { target: { value: 'micorreo@corporativo.com' } }
    );

    fireEvent.change(
      screen.getByPlaceholderText('Ingresa tu password'),
      { target: { value: 'mipass1234' } }
    );

    fireEvent.submit(
      screen.getByText('Login')
    )

    expect( screen.getByText('Calculadora') ).toBeInTheDocument();
    expect(container).toMatchSnapshot();

   })


});