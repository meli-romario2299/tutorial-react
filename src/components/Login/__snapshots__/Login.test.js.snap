// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente de Login Testeando la funcionalidad del formulario 1`] = `
<div>
  <div>
    Calculadora
  </div>
  <hr />
  <button>
    Sumar +1
  </button>
  <div>
    Resultado: 
    10
  </div>
  <button>
    Back
  </button>
</div>
`;

exports[`Prueba del componente de Login Verificar si el componente se encuentra renderizado 1`] = `
<div>
  <div
    id="login"
    style="background-color: red;"
  >
    View de Login
  </div>
  <form>
    <input
      name="email"
      placeholder="Ingrese su email corporativo"
      type="email"
      value="miemail@gmail.com"
    />
    <br />
    <input
      name="password"
      placeholder="Ingresa tu password"
      type="text"
      value="12345"
    />
    <br />
    <button
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;
